core-bundle/src/Security/Voter/DataContainer/FavoritesVoter.php

```diff
@@ -38,24 +35,27 @@ protected function getTable(): string
 
     protected function isGranted(CreateAction|DeleteAction|ReadAction|UpdateAction $action): bool
     {
-        return match (true) {
-            $action instanceof CreateAction => true,
-            $action instanceof ReadAction,
+        $user = $this->security->getUser();
+
+        if (!$user instanceof BackendUser) {
+            return false;
+        }
+
+        $userId = (int) $user->id;
+
+        $canAccessCurrent = match (true) {
             $action instanceof UpdateAction,
-            $action instanceof DeleteAction => $this->checkAccess($action),
+            $action instanceof ReadAction,
+            $action instanceof DeleteAction => (int) $action->getCurrent()['user'] === $userId,
+            default => true,
         };
-    }
-
-    private function checkAccess(DeleteAction|ReadAction|UpdateAction $action): bool
-    {
-        $user = $this->security->getUser();
-        $userId = $user instanceof BackendUser ? (int) $user->id : 0;
 
-        $createdBy = (int) $this->connection->fetchOne(
-            'SELECT user FROM tl_favorites WHERE id = :id',
-            ['id' => $action->getCurrentId()],
-        );
+        $canAccessNew = match (true) {
+            $action instanceof CreateAction,
+            $action instanceof UpdateAction => !isset($action->getNew()['user']) || $action->getNew()['user'] === $userId,
```

Comment:
```suggestion
            $action instanceof UpdateAction => !isset($action->getNew()['user']) || (int) $action->getNew()['user'] === $userId,
```

At least that is what you did in line 49.